src/Symfony/Component/Cache/LockRegistry.php

```diff
@@ -123,9 +122,8 @@ public static function compute(callable $callback, ItemInterface $item, bool &$s
                 }
                 // if we failed the race, retry locking in blocking mode to wait for the winner
                 $logger?->info('Item "{key}" is locked, waiting for it to be released', ['key' => $item->getKey()]);
-                flock($lock, \LOCK_SH);
+                $locked = flock($lock, \LOCK_SH);
```

Comment:
I think this might be a safer fix to check the value, to force it to retry from scratch. This is extremely unlikely anyway I guess, so rather play it safe than risking an inconsistent state below in the try/catch no?

```suggestion
                if (!flock($lock, \LOCK_SH)) {
                    continue;
                }
```